### PR TITLE
CORE-8903 Editing teams

### PR DESCRIPTION
**Pull request by ash3rz**
A lot of the extra changes in `EditTeamPresenterImpl` are due to wanting to organize all the methods in there a little better, so sorry for the clutter.

### CODE REVIEW

File: de.properties.tmpl [hunk 1]
```diff
@@ -214,7 +214,8 @@ org.iplantc.discoveryenvironment.redis.password = {{ key (printf "%s/de/redis-pa
 ###############################################################################
 # Groups Settings
 ###############################################################################
-org.iplantc.discoveryenvironment.groups.grouper-all-id = {{ key (printf "%s/de/grouper-id"  $base) }}
+org.iplantc.discoveryenvironment.groups.grouper-all-id = {{ key (printf "%s/de/grouper-all-id"  $base) }}
 org.iplantc.discoveryenvironment.groups.grouper-all-display-name = {{ key (printf "%s/de/grouper-all-display-name"  $base) }}
+org.iplantc.discoveryenvironment.groups.grouper-id = {{ key (printf "%s/de/grouper-id"  $base) }}
```
Comment (psarando) [inline de.properties.tmpl]: Don't forget: we didn't add this `de/grouper-id` key in cyverse-de/ansible#46, so now this config should be reading directly from `apps/grouper-user`.

File: de-lib/src/main/java/org/iplantc/de/client/models/collaborators/Subject.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ default String getSubjectDisplayName() {
         if (Strings.isNullOrEmpty(subjectName)) {
             subjectName = getFirstName() + " " + getLastName();
         }
-        if (!isGroup() || !hasGroupLongName(subjectName)) {
+        if (!isGroup() && !hasGroupLongName(subjectName)) {
             return subjectName;
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/client/models/collaborators/Subject.java]: Yeah, I need to clean this up - it is absolutely confusing.  This was my lazy way of fixing an issue I had and it makes no sense but it works and then I moved on to the next issue 😄 . 
I'm slightly tempted to ditch the `default` method idea and create a more robust util class that can handle the differences between a `Group` and a `Subject`, but I'm not sure yet.  I'll definitely refactor though.

File: de-lib/src/main/java/org/iplantc/de/client/services/GroupServiceFacade.java [hunk 3]
```diff
@@ -103,7 +112,15 @@
      * @param group
      * @param callback
      */
-    void updateGroup(String originalGroup, Group group, AsyncCallback<Group> callback);
+    void updateList(String originalGroup, Group group, AsyncCallback<Group> callback);
+
+    /**
+     * Update the details of a Collaborator List
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/client/services/GroupServiceFacade.java]: Oops, copy-paste error 😄

File: de.properties.tmpl (hunk 1 above)
Comment (ash3rz) [inline de.properties.tmpl]: Ah yes, good catch!

**Comment by ash3rz** [comment]
Thanks for the review!

**Pull request merged**